test/server/cards/02_SHD/events/Pillage.spec.ts

```diff
@@ -0,0 +1,92 @@
+
+describe('Pillage', function() {
+    integration(function(contextRef) {
+        describe('Pillage\'s ability', function() {
+            it('should let the player target the opponent, and let the opponent discard 2 cards from their hand', function() {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['pillage'],
+                    },
+                    player2: {
+                        hand: ['alliance-xwing', 'battlefield-marine', 'imperial-interceptor', 'wampa']
+                    }
+                });
+
+                const { context } = contextRef;
+                const { player1, player2, pillage, allianceXwing, battlefieldMarine, imperialInterceptor, wampa } = context;
+
+                player1.clickCard(pillage);
+                player1.clickPrompt('Opponent');
+                expect(player2).toBeAbleToSelectExactly([
+                    allianceXwing, battlefieldMarine, imperialInterceptor, wampa
+                ]);
+
+                player2.clickCard(imperialInterceptor);
+                player2.clickCard(wampa);
+                player2.clickPrompt('Done');
+
+                expect(wampa).toBeInLocation(Location.Discard);
+                expect(imperialInterceptor).toBeInLocation(Location.Discard);
+
+                expect(allianceXwing).toBeInLocation(Location.Hand);
+                expect(battlefieldMarine).toBeInLocation(Location.Hand);
+
+                expect(player2).toBeActivePlayer();
+            });
+
+            it('should let the player target the opponent, and automatically discard the only card in the opponents hand', function() {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['pillage'],
+                    },
+                    player2: {
+                        hand: ['imperial-interceptor']
+                    }
+                });
+
+                const { context } = contextRef;
+                const { player1, player2, pillage, imperialInterceptor } = context;
+
+                player1.clickCard(pillage);
+                player1.clickPrompt('Opponent');
+
+                expect(imperialInterceptor).toBeInLocation(Location.Discard);
+
+                expect(player2).toBeActivePlayer();
+            });
+
+            it('should let the player target the themselves, and let the player discard 2 cards from their hand', function() {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['pillage', 'alliance-xwing', 'battlefield-marine', 'imperial-interceptor', 'wampa'],
+                    },
+                });
+
+                const { context } = contextRef;
+                const { player1, player2, pillage, allianceXwing, battlefieldMarine, imperialInterceptor, wampa } = context;
```

Comment:
Not a suggestion per se but just a note - the test repo had a pattern like this originally and I moved us away from it b/c it's annoying to have to make a change in one more place if we change cards involved in the test. That said, it's highly personal what people find more and less annoying, so if you prefer declaring up front like this then go for it

Reply:
I prefer it, and some even consider it a best practice to destructure object properties that are use more than once to reduce code clutter and clearly denote what is used up front. AirBnb style guide is what my team follows (pretty popular outside my team as well) is probably why I picked up this habit, and they recommend it and even enforce it in their eslint plugins: https://github.com/airbnb/javascript?tab=readme-ov-file#destructuring--object

Of course this is just a style preference so totally just a preference thing at the end of the day. Both are functionally equivalent.